packages/workflow/src/features/correction/routes.ts

```diff
@@ -293,28 +298,56 @@ export const routes = [
       const { correction, ...registration } =
         recordInputWithUploadedAttachments.registration!
 
+      /*
+       * Separate tasks from the record
+       */
+      const [correctionRequestApprovedTask, correctedTask] =
+        getTasksInAscendingOrder(recordInCorrectedState)
+      const recordWithoutTasks = getRecordWithoutTasks(recordInCorrectedState)
+
+      /*
+       * Create and update the valid record that only has one task (the corrected task)
+       */
+      const newRecord = addTaskToRecord(recordWithoutTasks, correctedTask)
+
+      newRecord satisfies ValidRecord
+
       const recordWithUpdatedValues = updateFHIRBundle(
-        recordInCorrectedState,
+        newRecord,
         { ...recordInputWithUploadedAttachments, registration },
         getEventType(record)
       )
 
       /*
        * Saves the previous task in the "approved" state and the new task (record put to Registered state)
+       * Note that because we are storing the task "twice", the approved task needs to be first in the list!
        */
 
-      await sendBundleToHearth(recordWithUpdatedValues)
-
-      const recordWithOnlyTheNewTask = withOnlyLatestTask(
+      const updatedCorrectedTask = getTaskFromSavedBundle(
         recordWithUpdatedValues
       )
+      const recordWithNoTasks = getRecordWithoutTasks(recordWithUpdatedValues)
+
+      // First add the approved task to the record
+      const recordWithApprovedTask = addTaskToRecord(
+        recordWithNoTasks,
+        correctionRequestApprovedTask
+      )
+
+      // Then add the corrected task to the record
+      const fullBundleToBeSaved = addTaskToRecord(
+        recordWithApprovedTask,
+        updatedCorrectedTask
+      )
+
+      await sendBundleToHearth(fullBundleToBeSaved)
 
       /*
        * Create metrics events & reindex the bundle in elasticsearch
        */
 
-      await createNewAuditEvent(recordWithOnlyTheNewTask, token)
-      await indexBundle(recordWithOnlyTheNewTask, token)
+      await createNewAuditEvent(newRecord, token)
+      await indexBundle(newRecord, token)
```

Comment:
Shouldn't we send the `recordWithUpdatedValues` to the other services? As new newRecord still has the old values and not the corrected ones if I'm not wrong

Reply:
Great find, thanks!